lib/facter/java_path.rb

```diff
@@ -0,0 +1,19 @@
+# Fact: java_path
+#
+# Purpose: get absolute path of java executable
+#
+# Resolution:
+#   Uses `readlink` to resolve the path of `/usr/bin/java`
+#
+# Caveats:
+#   Requires readlink
+#
+# Notes:
+#   None
+Facter.add(:java_path) do
+  setcode do
+    if Facter::Util::Resolution.which('readlink')
+      Facter::Util::Resolution.exec('readlink -e /usr/bin/java').strip
```

Comment:
this will not work on [Darwin](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man1/readlink.1.html), or [FreeBSD](https://www.freebsd.org/cgi/man.cgi?readlink) or [OpenBSD](http://www.openbsd.org/cgi-bin/man.cgi/OpenBSD-current/man1/readlink.1?query=readlink), or solaris, where it doesn't exist at all…